src/herbivoresAndCarnivores.js

```diff
@@ -1,16 +1,55 @@
 'use strict';
 
 class Animal {
-  // write your code here
-}
+  // static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  };
+
+  decreaseHealth(amount) {
+    this.health -= amount;
+
+    if (this.health <= 0) {
+      this.die();
+    }
+  };
+
+  die() {
+    const index = Animal.alive.indexOf(this);
+
+    if (index !== -1) {
+      Animal.alive.splice(index, 1);
+    }
+  };
+};
 
 class Herbivore extends Animal {
-  // write your code here
-}
+  constructor(name, health = 100) {
+    super(name, health);
+    this.hidden = false;
+  };
+
+  hide() {
+    this.hidden = true;
+  };
+};
 
 class Carnivore extends Animal {
-  // write your code here
-}
+  constructor(name, health = 100) {
+    super(name, health);
+  };
+
+  bite(target) {
+    if (target instanceof Herbivore && target.hidden !== true) {
```

Comment:
```suggestion
    if (target instanceof Herbivore && !target.hidden) {
```

same but shorter

Reply:
I know, but it didn't work when I first wrote it

Reply:
must work if you did everything correct